cli/src/commands/git/push.rs

```diff
@@ -279,36 +279,59 @@ pub fn cmd_git_push(
             &tx.base_workspace_helper().revset_parse_context(),
         )?)
         .range(&RevsetExpression::commits(new_heads));
-    for commit in tx
-        .base_workspace_helper()
-        .attach_revset_evaluator(commits_to_push)?
-        .evaluate_to_commits()?
+
```

Comment:
nit: you can also move `new_heads`/`old_heads`/`commits_to_push` inside the block. It might be better to extract a check function because `cmd_git_push()` is super long.

Reply:
Created `validate_commits_ready_to_push`